### PR TITLE
fix: prevent eth sendtransaction

### PR DESCRIPTION
**Pull request by tynes**
**Description**
This PR blocks certain endpoints from being available for users based on the environment variable `USING_OVM`. We need to turn this env var into an actual flag and have it select a `consensus.Engine`, but will leave that for future work. The blocked endpoints are:
- `eth_fillTransaction`
- `eth_sign`
- `eth_sendTransaction`
- `eth_signTransaction`

It also adds in the `txmeta.rawTransaction` field in an additional place where transactions are serialized based on user input. This code path should no longer be reachable because it is behind `eth_sendTransaction`.

### CODE REVIEW

File: l2geth/internal/ethapi/api.go [hunk 1]
```diff
@@ -1723,6 +1724,9 @@ func (s *PublicTransactionPoolAPI) SendTransaction(ctx context.Context, args Sen
 // FillTransaction fills the defaults (nonce, gas, gasPrice) on a given unsigned transaction,
 // and returns it to the caller for further processing (signing + broadcast)
 func (s *PublicTransactionPoolAPI) FillTransaction(ctx context.Context, args SendTxArgs) (*SignTransactionResult, error) {
+	if vm.UsingOVM {
+		return nil, errors.New("Cannot fill transaction")
```
Comment (snario) [inline l2geth/internal/ethapi/api.go:1728]: Should have some kind of prefix e.g., "OVMError: Cannot fill transaction" or in some other way indicate this error is thrown to indicate it is due to the vm being wrong

**Review by snario — APPROVED** [review]

**Comment by gakonst** [comment]
Should this PR not be against master?

**Comment by smartcontracts** [comment]
Intuitively it seems like we should be able to block these endpoints via some sort of configuration. Is that not the case?

**Comment by gakonst** [comment]
@smartcontracts no, you enable entire namespaces in geth, but not block off specific methods.

File: l2geth/internal/ethapi/api.go (hunk 1 above)
Comment (gakonst) [reply l2geth/internal/ethapi/api.go:1728]: IMO just create a global error called: `var errOvmNotSupported = errors.New("OVMError: Endpoint disabled")`, maybe wrapping it with some additional context per method, although not necessary.

File: .changeset/kind-plants-tickle.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@eth-optimism/l2geth': patch
+---
+
+Block access to RPCs related to signing transactions
```
Comment (gakonst) [inline .changeset/kind-plants-tickle.md:5]: Changeset should include the info about the raw rlp tx being encoded.

**Review by gakonst — CHANGES_REQUESTED** [review]

**Comment by tynes** [comment]
> Intuitively it seems like we should be able to block these endpoints via some sort of configuration instead of by modifying geth. Is that not the case?

Its possible to run a proxy in front of geth that blocks certain RPC methods

**Pull request merged**